src/main/groovy/pl/allegro/tech/build/axion/release/infrastructure/git/GitRepository.groovy

```diff
@@ -168,6 +169,50 @@ class GitRepository implements ScmRepository {
                 repository.branch.current.name
         )
     }
+		
+		boolean isHeadCommit(String commitId) {
+			if(commitId?.trim()) {
+				ObjectId headId = repository.repository.jgit.repository.resolve(Constants.HEAD)
+				return Objects.equals(ObjectId.fromString(commitId), headId)
+			}
+			return false;
+		}
+		
+		LinkedHashMap<String, List<String>> allTaggedCommits(Pattern pattern, String maybeSinceCommit, boolean inclusive) {
+			LinkedHashMap<String, List<String>> allTaggedCommits = new ArrayList<>();
+			if (!hasCommits()) {
+					return allTaggedCommits
+			}
+
+			Map<String, List<String>> allTags = tagsMatching(pattern)
+
+			ObjectId headId = repository.repository.jgit.repository.resolve(Constants.HEAD)
+
+			ObjectId startingCommit;
+			if (maybeSinceCommit != null) {
+					startingCommit = ObjectId.fromString(maybeSinceCommit)
+			} else {
+					startingCommit = headId
+			}
+
+			RevWalk walk = walker(startingCommit)
+			if (!inclusive) {
+					walk.next()
+			}
+
+			List tagsList = null
+
+			RevCommit currentCommit
```

Comment:
i see the logic of this method is very much the same as `latestTagsInternal`. so maybe you could create internal method like `List<TagsOnCommit> retrieveTags(boolean stopOnFirstTag)` which could be used by both `latestTags` and `allTaggedCommits` ? they just differ in stopping point definition